File: Year02/2nd semester/MPP/Lab/mpp-proiect-csharp-AlexandruPaul21/CsharpProj/service/Service.cs
using CsharpProj.model;
using CsharpProj.repository;
// ReSharper disable All

namespace CsharpProj.service;

public class Service
{
    public IAgencyRepository AgencyRepository {get; set;}
    public IBookingRepository BookingRepository { get; set; }
    public IClientRepository ClientRepository { get; set; }
    public IFlightRepository FlightRepository { get; set; }

    public Service(IAgencyRepository agencyRepository, IBookingRepository bookingRepository, IClientRepository clientRepository, IFlightRepository flightRepository)
    {
        AgencyRepository = agencyRepository;
        BookingRepository = bookingRepository;
        ClientRepository = clientRepository;
        FlightRepository = flightRepository;
    }
    
    public bool validateLogin(string username, string password)
    {
        return AgencyRepository.existsUser(username, password);
    }
    
    public IEnumerable<Flight> findFlightByDestinationAndDate(string destination, DateTime dateTime)
    {
        return FlightRepository.findFlightByDestinationAndDate(destination, dateTime);
    }
    
    public int getNumberOfBookingsForFlight(Flight flight)
    {
        return BookingRepository.getNumberOfBookingsForFlight(flight);
    }
    
    public IEnumerable<Flight> getAllFlights()
    {
        return FlightRepository.findAll();
    }

    public bool bookFlight(long flightId, string clientName, string clientAddress, List<string> clients)
    {
        Client client = new Client(clientName, clientAddress);
        client.Id = ClientRepository.getLowestAvailableId() + 1;
        ClientRepository.save(client);
        Booking booking = new Booking(FlightRepository.findOne(flightId), client, clients);
        booking.Id = BookingRepository.getLowestAvailableId() + 1;
        return BookingRepository.save(booking) != null;
    }

}

File: Year02/2nd semester/MPP/Lab/mpp-proiect-java-AlexandruPaul21/JavaFxProject/src/main/java/com/example/javafxproject/service/Service.java
package com.example.javafxproject.service;

import com.example.javafxproject.model.Booking;
import com.example.javafxproject.model.Client;
import com.example.javafxproject.model.Flight;
import com.example.javafxproject.repository.*;

import java.time.LocalDateTime;
import java.util.List;

public class Service {
    private final IAgencyRepository agencyRepository;
    private final IBookingRepository bookingRepository;
    private final IClientRepository clientRepository;
    private final IFlightRepository flightRepository;

    public Service(IAgencyRepository agencyRepository, IBookingRepository bookingRepository, IClientRepository clientRepository, IFlightRepository flightRepository) {
        this.agencyRepository = agencyRepository;
        this.bookingRepository = bookingRepository;
        this.clientRepository = clientRepository;
        this.flightRepository = flightRepository;
    }

    public boolean validateLogin(String username, String password) {
        return agencyRepository.existsUser(username, password);
    }

    public Iterable<Flight> getFlightsForDestinationAndDate(String destination, LocalDateTime date) {
        return flightRepository.findFlightByDestinationAndDate(destination, date);
    }

    public int getNumberOfBookingsForFlight(Long id) {
        return bookingRepository.getNumberOfBookingsForFlight(id);
    }

    public Iterable<Flight> getAllFlights() {
        return flightRepository.findAll();
    }

    public boolean bookFlight(Long flightId, String clientName, String clientAddress, List<String> clients) {
        Client client = new Client(clientName, clientAddress);
        client.setId(clientRepository.getLowestAvbId() + 1);
        clientRepository.save(client);
        Booking booking = new Booking(flightRepository.findOne(flightId), client, clients);
        booking.setId(bookingRepository.getLowestAvbId() + 1);
        return bookingRepository.save(booking) != null;
    }
}
